src/x11/screen.c

```diff
@@ -82,27 +82,33 @@ void alloc_screen_ar(int n)
 
 int randr_event_base = 0;
 
+static int randr_major_version = 1;
```

Comment:
Why default these to 1 and 5? Just leave them uninitialized (since they're globals they're guaranteed to be 0).

Reply:
This was me reading the protocol docs which say that the client has to send the version it expects, but the xrandr code sends this itself, it doesn't use the initial value of the pointers you pass in. I'll fix this.

Reply:
> This was me reading the protocol docs which say that the client has to send the version it expects

Where does it say that? It's not mentioned anywhere in the xrandr man page?

Reply:
It's in the protocol specification, but the library API doesn't match that.